### PR TITLE
Tool install cleanup

### PR DESCRIPTION
**Pull request by Forgind**
This was mostly just taking various suggestions from analyzers, but I also refactored NuGetPackageDownloader's LoadNuGetSources method because it was bothering me that if there are source feed overrides, it still calculates default sources then just throws them away and GetUserFacingMessages because I like yield returns.

### CODE REVIEW

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs [hunk 1]
```diff
@@ -367,62 +387,28 @@ private IEnumerable<PackageSource> LoadNuGetSources(PackageId packageId, Package
...
-            {
-                retrievedSources = customSources;
-            }
-            else
-            {
-                retrievedSources = defaultSources;
-            }
+        private IEnumerable<PackageSource> LoadNuGetSources(PackageId packageId, PackageSourceLocation packageSourceLocation = null, PackageSourceMapping packageSourceMapping = null)
+        {
+            var sources = packageSourceLocation != null && packageSourceLocation.SourceFeedOverrides.Any() ?
```
Comment (MiYanni) [inline src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:395]: nit: Purely subjective thing. I don't remember if the parens are needed or not.

```suggestion
            var sources = (packageSourceLocation?.SourceFeedOverrides.Any() ?? false) ?
```

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs [hunk 2]
```diff
@@ -808,12 +784,13 @@ private async Task<IEnumerable<string>> GetPackageIdsForSource(AutoCompleteResou
...
-            if (!_sourceRepositories.ContainsKey(source))
+            if (!_sourceRepositories.TryGetValue(source, out SourceRepository value))
             {
-                _sourceRepositories.Add(source, Repository.Factory.GetCoreV3(source));
+                value = Repository.Factory.GetCoreV3(source);
+                _sourceRepositories.Add(source, value);
             }
 
-            return _sourceRepositories[source];
+            return value;
```
Comment (MiYanni) [inline src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:793]: I like this change! 😊

File: src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs [hunk 3]
```diff
@@ -23,16 +22,11 @@ public ToolInstallCommand(
...
         {
-            _toolInstallLocalCommand =
-                toolInstallLocalCommand
-                ?? new ToolInstallLocalCommand(_parseResult);
+            _toolInstallLocalCommand = toolInstallLocalCommand;
 
-            _toolInstallGlobalOrToolPathCommand =
-                toolInstallGlobalOrToolPathCommand
-                ?? new ToolInstallGlobalOrToolPathCommand(_parseResult);
+            _toolInstallGlobalOrToolPathCommand = toolInstallGlobalOrToolPathCommand;
```
Comment (MiYanni) [inline src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs:27]: I'm actually a bit confused why this was changed to no longer initialize the fields, but instead, changed to only `new` a command within `Execute` instead. I don't see the benefit in the change, but maybe I'm missing something... 🤔

**Review by MiYanni — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs (hunk 3 above)
Comment (Forgind) [reply src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs:27]: Previously, we were eagerly initializing the commands, but we only actually need one of them...why make two if you're only going to use one? Doing it in Execute means we know which we need, and we don't have to make the other.

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs (hunk 1 above)
Comment (Forgind) [reply src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:395]: Not needed, but I don't have a strong preference

File: src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs (hunk 3 above)
Comment (MiYanni) [reply src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs:27]: Ah, makes sense.

**Pull request merged**